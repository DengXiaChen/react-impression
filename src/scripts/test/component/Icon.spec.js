import React from 'react';
import { shallow } from 'enzyme';
import { Icon } from '../../components/impression';

const setup = () => {
    const props = {
        type: 'arrow',
        size: 'lg',
        right: true,
    };

    const wrapper = shallow(<Icon {...props} />);

    return {
        props,
        wrapper,
    };
};

describe('Icon', () => {
    const { wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('i.fa.fa-arrow.fa-lg.offset-l').exists()).toBe(true);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
