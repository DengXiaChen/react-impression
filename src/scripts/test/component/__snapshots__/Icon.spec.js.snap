// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <i
    className="fa fa-arrow fa-lg offset-l"
/>,
  "nodes": Array [
    <i
      className="fa fa-arrow fa-lg offset-l"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Icon
        left={false}
        right={true}
        size="lg"
        type="arrow"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Icon {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "left": false,
          "right": true,
          "size": "lg",
          "type": "arrow",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <i
          className="fa fa-arrow fa-lg offset-l"
/>,
        "_debugID": 2,
        "_renderedOutput": <i
          className="fa fa-arrow fa-lg offset-l"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Icon
    left={false}
    right={true}
    size="lg"
    type="arrow"
/>,
}
`;
